ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -1445,6 +1506,73 @@ static bool ggml_vk_matmul_shmem_support(const vk_device& device, const std::vec
     return supported;
 }
 
+struct GpuPipelineConfig {
+    // GPU architecture identifier.
+    // Example: vk_device_architecture::AMD_GCN
+    vk_device_architecture arch;
+
+    // Mapping of pipeline names to their specific subgroup sizes.
+    // Example: {"soft_max_f32", 64}
+    std::unordered_map<std::string, uint32_t> pipelines;
+
+    // Default subgroup size for this GPU.
+    // Defaults to 0 if not explicitly provided.
+    uint32_t default_subgroup_size = 0;
+};
+
+// Common pipeline configuration for RDNA GPUs.
+static const std::unordered_map<std::string, uint32_t> rdna_common_pipelines = {
+    {"soft_max", 64}, {"im2col", 64},
+};
+
+// RDNA1 pipeline configuration.
+static std::unordered_map<std::string, uint32_t> rdna1_pipelines = rdna_common_pipelines;
+static const bool rdna1_initialized = (rdna1_pipelines.insert({
+    {"argmax", 64}, {"mul_mat_vec", 64},
+    {"mul_mat_vec_f16", 32}, {"mul_mat_vec_f32_f16", 32}
+}), true);
```

Comment:
This is an odd workaround to copy and add to a vector. Since it's only about 2 entries for now, let's just duplicate if necessary.